src/transformers/quantizers/quantizer_gptq.py

```diff
@@ -44,18 +44,29 @@ class GptqHfQuantizer(HfQuantizer):
 
     def __init__(self, quantization_config: QuantizationConfigMixin, **kwargs):
         super().__init__(quantization_config, **kwargs)
+
+        if not is_optimum_available():
+            raise ImportError(
+                "Loading a GPTQ quantized model requires optimum (`pip install optimum`)"
+            )
         from optimum.gptq import GPTQQuantizer
 
         self.optimum_quantizer = GPTQQuantizer.from_dict(self.quantization_config.to_dict_optimum())
 
     def validate_environment(self, *args, **kwargs):
+        if not is_optimum_available():
+            raise ImportError(
+                "Loading a GPTQ quantized model requires optimum (`pip install optimum`)"
+            )
+
+        if not is_auto_gptq_available():
+            raise ImportError(
+                "Loading a GPTQ quantized model requires the auto-gptq library (`pip install auto-gptq`)"
+            )
+
```

Comment:
Changed the order here so we can guarantee that the metadata check doesn't raise an error itself. Also separated but that's more of a preference of mine to differentiate, lmk if I should change this back to checking both at once.

Reply:
I think it's fine to have the checks separated, wdyt @SunMarc I think it's fine to have the checks separated, wdyt @SunMarc

Reply:
yes !